pkg/pillar/cmd/zedrouter/flowstats.go

```diff
@@ -0,0 +1,738 @@
+		if len(instData.intfAddrs) > 0 {
+			ipFlow.dbg1 = 6
+			if !isInMySubnets(entry.Forward.SrcIP, instData.intfAddrs) { // forw src is remote endpoint
+				ipFlow.dbg1 = 7
+				// in this case, if the forw.src is remote, then assume the reverse.src is the app
+				backSrcApp = true
+			} else if !isInMySubnets(entry.Reverse.SrcIP, instData.intfAddrs) {
+				ipFlow.dbg1 = 8
+				forwSrcApp = true
+			} else if !isInMySubnets(entry.Forward.DstIP, instData.intfAddrs) {
+				ipFlow.dbg1 = 9
+				backDstApp = true
+			} else if !isInMySubnets(entry.Reverse.DstIP, instData.intfAddrs) {
+				ipFlow.dbg1 = 10
+				forwDstApp = true
+			}
+		}
+	}
+
+	// Assume we know which one of the 4 IP addresses is the app, then we know
+	// which 'remote' IP address is in the flow tuple. Assign the Src/Dst and Ports
+	// If the App initiated the traffic, then the forw.src is the App IP, or the reverse.dst
+	// is the App IP
+	if forwSrcApp {
+		ipFlow.dbg2 = 1
+		ipFlow.SrcIP = entry.Forward.SrcIP
+		ipFlow.DstIP = entry.Reverse.SrcIP
+		ipFlow.SrcPort = entry.Forward.SrcPort
+		ipFlow.DstPort = entry.Reverse.DstPort
+		ipFlow.AppInitiate = true
+	} else if forwDstApp {
+		ipFlow.dbg2 = 2
+		ipFlow.DstIP = entry.Forward.DstIP
+		ipFlow.SrcIP = entry.Reverse.DstIP
+		ipFlow.DstPort = entry.Forward.DstPort
+		ipFlow.SrcPort = entry.Reverse.DstPort
+	} else if backSrcApp {
+		ipFlow.dbg2 = 3
+		ipFlow.SrcIP = entry.Forward.SrcIP
+		ipFlow.DstIP = entry.Reverse.SrcIP
+		ipFlow.SrcPort = entry.Forward.SrcPort
+		ipFlow.DstPort = entry.Reverse.SrcPort
+	} else if backDstApp {
+		ipFlow.dbg2 = 4
+		ipFlow.SrcIP = entry.Reverse.DstIP
+		ipFlow.DstIP = entry.Forward.DstIP
+		ipFlow.SrcPort = entry.Reverse.DstPort
+		ipFlow.DstPort = entry.Forward.DstPort
+		ipFlow.AppInitiate = true
+	} else { // if we can not find our App endpoint is part of the flow, something is wrong
+		ipFlow.dbg2 = 5
+		log.Infof("FlowStats: flow entry can not locate app IP address, appNum %d, %s", AppNum, entry.String())
+		return ipFlow
+	}
+
+	ipFlow.foundApp = true
+	// If App initiated traffic, forw is sending 'OUT', otherwise forw is receiving 'IN'
+	if ipFlow.AppInitiate {
+		ipFlow.SendPkts = entry.Forward.Packets
+		ipFlow.SendBytes = entry.Forward.Bytes
+		ipFlow.RecvPkts = entry.Reverse.Packets
+		ipFlow.RecvBytes = entry.Reverse.Bytes
+	} else {
+		ipFlow.SendPkts = entry.Reverse.Packets
+		ipFlow.SendBytes = entry.Reverse.Bytes
+		ipFlow.RecvPkts = entry.Forward.Packets
+		ipFlow.RecvBytes = entry.Forward.Bytes
+	}
+	return ipFlow
+}
+
+func checkAppIPAddr(appAddrs []net.IP, entryIP net.IP) bool {
+	for _, appIP := range appAddrs {
+		if strings.Compare(appIP.String(), entryIP.String()) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// check to see if the IP address is on my local subnets
+func isInMySubnets(faddr net.IP, addrList []net.Addr) bool {
+	for _, address := range addrList {
+		if ipnet, ok := address.(*net.IPNet); ok {
+			prefix := faddr.Mask(ipnet.Mask)
```

Comment:
Why not use https://golang.org/pkg/net/#IPNet.Contains for this?

Reply:
I'll try it, will see if Contains work.